skyportal/handlers/api/reminder.py

```diff
@@ -0,0 +1,796 @@
+            url_endpoint = "/shifts"
+            action = 'skyportal/REFRESH_REMINDER_SHIFT'
+            payload = {'id': resource_id}
+        else:
+            return self.error(f'Unknown resource type "{associated_resource_type}".')
+
+        ws_flow = Flow()
+        for user in users:
+            DBSession().add(
+                UserNotification(
+                    user=user,
+                    text=text_to_send,
+                    notification_type="mention",
+                    url=url_endpoint,
+                )
+            )
+            ws_flow.push(user.id, "skyportal/FETCH_NOTIFICATIONS")
+
+        self.verify_and_commit()
+
+        self.push_all(action, payload)
+        return self.success(
+            data={'reminder_ids': [reminder.id for reminder in reminders]}
+        )
+
+    @permissions(['Reminder'])
+    def put(self, associated_resource_type, resource_id, reminder_id):
+        """
+        ---
+        description: Update a reminder
+        tags:
+          - reminders
+        parameters:
+          - in: path
+            name: associated_resource_type
+            required: true
+            schema:
+              type: string
+              enum: [sources, spectrum, gcn_event, shift]
+            description: |
+               What underlying data the reminder is on:
+               "sources" or "spectra" or "gcn_event" or "shift".
+          - in: path
+            name: resource_id
+            required: true
+            schema:
+              type: string
+              enum: [sources, spectra, gcn_event, shift]
+            description: |
+               The ID of the source or spectrum
+               that the reminder is posted to.
+               This would be a string for an object ID
+               or an integer for a spectrum, gcn_event or shift.
+          - in: path
+            name: reminder_id
+            required: true
+            schema:
+              type: integer
+        requestBody:
+          content:
+            application/json:
+              schema:
+                allOf:
+                  - $ref: '#/components/schemas/ReminderNoID'
+                  - type: object
+                    properties:
+                      group_ids:
+                        type: array
+                        items:
+                          type: integer
+                        description: |
+                          List of group IDs corresponding to which groups should be
+                          able to view reminder.
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+
+        try:
+            reminder_id = int(reminder_id)
+        except (TypeError, ValueError):
+            return self.error("Must provide a valid (scalar integer) reminder ID. ")
```

Comment:
Since this only addresses existing reminders, it might be better to use PATCH instead of PUT.